spec/openapi.yaml

```diff
@@ -120,6 +120,65 @@ paths:
         '500':
           description: An unexpected error occured.
 
+  /tag/metadata:
+    post:
+      operationId: assign-tag-metadata
+      summary: Assign metadata to an existing tag.
+      description: >
+        You can assign metadata to a tag.  This information is provided as a name-value (key-value) pair.  In the case of providing any name-value where the name is already assigned as metadata to the tag, the value will be updated with value supplied.
+        To delete/remove a name-value pair stored as metadata on a tag, you must use the `delete-tag-metadata` operation.
+      tags:
+        - Tags
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/UpdateTagMetadataRequest'
```

Comment:
small nit - if operation called `Assign` then schema object should use same verb (replacing of `Update`), or vice-versa